File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Isolator Bricklet

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletIsolator i = new BrickletIsolator(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current statistics
		long messagesFromBrick, messagesFromBricklet;
		     int connectedBrickletDeviceIdentifier; string connectedBrickletUID;
		i.GetStatistics(out messagesFromBrick, out messagesFromBricklet,
		                out connectedBrickletDeviceIdentifier, out connectedBrickletUID);

		Console.WriteLine("Messages From Brick: " + messagesFromBrick);
		Console.WriteLine("Messages From Bricklet: " + messagesFromBricklet);
		Console.WriteLine("Connected Bricklet Device Identifier: " + connectedBrickletDeviceIdentifier);
		Console.WriteLine("Connected Bricklet UID: " + connectedBrickletUID);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, isolator_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Isolator Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let i = IsolatorBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current statistics.
    let statistics = i.get_statistics().recv()?;

    println!("Messages From Brick: {}", statistics.messages_from_brick);
    println!(
        "Messages From Bricklet: {}",
        statistics.messages_from_bricklet
    );
    println!(
        "Connected Bricklet Device Identifier: {}",
        statistics.connected_bricklet_device_identifier
    );
    println!(
        "Connected Bricklet UID: {}",
        statistics.connected_bricklet_uid
    );

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
